autotest/ogr/ogr_gml.py

```diff
@@ -4644,3 +4645,236 @@ def test_ogr_gml_force_opening(tmp_vsimem):
+        schema = None
+        is_path = False
+
+    with gdal.quiet_errors():
+
+        if is_path:
+            schema_path = tmp_path / "ogr_schema.json"
+            with open(schema_path, "w+") as f:
+                f.write(schema)
+            open_options.append("OGR_SCHEMA=" + str(schema_path))
+        elif schema:
+            open_options.append("OGR_SCHEMA=" + schema)
+        else:
+            open_options = []
+
+        # Validate the JSON schema
+        if not expected_warning and schema:
+            schema = json.loads(schema)
+            gdaltest.validate_json(schema, "ogr_fields_override.schema.json")
+
+        # Check error if expected_field_types is empty
+        if not expected_field_types:
+            ds = gdal.OpenEx(
+                tmp_path / "test_point.gml",
+                gdal.OF_VECTOR | gdal.OF_READONLY,
+                open_options=open_options,
+                allowed_drivers=["GML"],
+            )
+            assert (
+                gdal.GetLastErrorMsg().find(expected_warning) != -1
+            ), f"Warning {expected_warning} not found, got {gdal.GetLastErrorMsg()} instead"
+            assert ds is None
+        else:
+
+            with gdal.OpenEx(
+                tmp_path / "test_point.gml",
+                gdal.OF_VECTOR | gdal.OF_READONLY,
+                open_options=open_options,
+                allowed_drivers=["GML"],
+            ) as gml_ds:
+
+                assert gml_ds.GetLayerCount() == 1, "wrong number of layers"
+
+                lyr = gml_ds.GetLayerByName("test_point")
+
+                feat = lyr.GetNextFeature()
+
+                if len(expected_field_names) == 0:
+                    expected_field_names = ("fid", "dbl", "int", "str")
+
+                # Check field types fid, dbl, int, str
+                for i in range(len(expected_field_names)):
+                    try:
+                        expected_str_type, expected_str_subtype = expected_field_types[
+                            i
+                        ]
+                        assert feat.GetFieldDefnRef(i).GetType() == expected_str_type
+                        assert (
+                            feat.GetFieldDefnRef(i).GetSubType() == expected_str_subtype
+                        )
+                    except TypeError:
+                        expected_str_type = expected_field_types[i]
+                        assert feat.GetFieldDefnRef(i).GetType() == expected_str_type
+
+                # Test width and precision override
+                if len(open_options) > 0 and "precision" in open_options[0]:
+                    assert feat.GetFieldDefnRef(2).GetWidth() == 7
+                    assert feat.GetFieldDefnRef(2).GetPrecision() == 3
+
+                if expected_warning:
+                    assert (
+                        gdal.GetLastErrorMsg().find(expected_warning) != -1
+                    ), f"Warning {expected_warning} not found, got {gdal.GetLastErrorMsg()} instead"
```

Comment:
It would be appropriate to also fetch a feature and check its content, to ensure that the override of field types is fully taken into account in the driver

Reply:
done